app/models/mail_handler.rb

```diff
@@ -161,8 +161,8 @@ def receive_issue
     end
 
     issue = WorkPackage.new(author: user, project: project)
-    issue.safe_attributes = issue_attributes_from_keywords(issue)
-    issue.safe_attributes = { 'custom_field_values' => custom_field_values_from_keywords(issue) }
+    issue.attributes = issue_attributes_from_keywords(issue)
```

Comment:
Yes, `PermittedParams` as defined by OP gets access to the user instance, so we can do permission logic there, which is an improvement over doing it in the model (most of which I deleted). 

The `safe_attributes=` is our own implementation, not the one from Redmine, so I left it there as it is doing more work (but I deleted where it was used, because i conflicted it with the Redmine one), so how about I restore it there? This is even simpler than your solutions (which are good so we can do them in a different package).